### ISSUE TITLE
1.0.3 Pre-built binaries not found for v64-linux-x64

### ISSUE THREAD

**Issue by lajpatshah** [kelektiv/node.bcrypt.js#677]
Hi 

Till last week below binaries were available. Were these binaries removed recently?

Do you have a specific timeline till which binaries are kept in the system after which it is deleted? Some idea on this can help so that we also prepare our self with new version if need be.

node-pre-gyp ERR! Tried to download(undefined): https://github.com/kelektiv/node.bcrypt.js/releases/download/v1.0.3/bcrypt_lib-v1.0.3-node-v64-linux-x64.tar.gz 
node-pre-gyp ERR! Pre-built binaries not found for bcrypt@1.0.3 and node@10.13.0 (node-v64 ABI) (falling back to source compile with node-gyp)

**Comment by lajpatshah** [comment]
We will surely migrate to new version but meanwhile I am still facing below issue. can you help here since my site is live and i cannot migrate without testing all my packages .

I am running on AWS linux and I am getting below error for built


Application deployment failed at 2018-11-19T12:48:12Z with exit status 1 and error: Hook /opt/elasticbeanstalk/hooks/appdeploy/pre/50npm.sh failed.+ /opt/elasticbeanstalk/containerfiles/ebnode.py --action npm-installnpm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but npm-shrinkwrap.json was generated for lockfileVersion@0. I'll try to do my best with it!npm WARN deprecated bcrypt@1.0.3: bcrypt < v2.0.0 is susceptible to bcrypt wrap-around bug. Upgrade to bcrypt >= v2.0.0 for improved support for newer bcrypt hashesnpm WARN deprecated hoek@2.16.3: This version is no longer maintained. Please upgrade to the latest version.npm WARN deprecated joi@6.10.1: This version is no longer maintained. Please upgrade to the latest version.npm WARN deprecated topo@1.1.0: This version is no longer maintained. Please upgrade to the latest version.> bcrypt@1.0.3 install /tmp/deployment/application/node_modules/bcrypt> node-pre-gyp install --fallback-to-buildnode-pre-gyp ERR! Tried to download(404): https://github.com/kelektiv/node.bcrypt.js/releases/download/v1.0.3/bcrypt_lib-v1.0.3-node-v64-linux-x64.tar.gz node-pre-gyp ERR! Pre-built binaries not found for bcrypt@1.0.3 and node@10.13.0 (node-v64 ABI) (falling back to source compile with node-gyp) node-pre-gyp ERR! Tried to download(undefined): https://github.com/kelektiv/node.bcrypt.js/releases/download/v1.0.3/bcrypt_lib-v1.0.3-node-v64-linux-x64.tar.gz node-pre-gyp ERR! Pre-built binaries not found for bcrypt@1.0.3 and node@10.13.0 (node-v64 ABI) (falling back to source compile with node-gyp) make: Entering directory `/tmp/deployment/application/node_modules/bcrypt/build'CXX(target) Release/obj.target/bcrypt_lib/src/blowfish.omake: Entering directory `/tmp/deployment/application/node_modules/bcrypt/build'CXX(target) Release/obj.target/bcrypt_lib/src/blowfish.oCXX(target) Release/obj.target/bcrypt_lib/src/bcrypt.osed: can't read ./Release/.deps/Release/obj.target/bcrypt_lib/src/blowfish.o.d.raw: No such file or directoryrm: cannot remove ‘./Release/.deps/Release/obj.target/bcrypt_lib/src/blowfish.o.d.raw’: No such file or directorymake: *** [Release/obj.target/bcrypt_lib/src/blowfish.o] Error 1make: Leaving directory `/tmp/deployment/application/node_modules/bcrypt/build'gyp ERR! build error gyp ERR! stack Error: `make` failed with exit code: 2gyp ERR! stack at ChildProcess.onExit (/opt/elasticbeanstalk/node-install/node-v10.13.0-linux-x64/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)gyp ERR! stack at ChildProcess.emit (events.js:182:13)gyp ERR! stack at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)gyp ERR! System Linux 4.14.77-69.57.amzn1.x86_64gyp ERR! command "/opt/elasticbeanstalk/node-install/node-v10.13.0-linux-x64/bin/node" "/opt/elasticbeanstalk/node-install/node-v10.13.0-linux-x64/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/tmp/deployment/application/node_modules/bcrypt/lib/binding/bcrypt_lib.node" "--module_name=bcrypt_lib" "--module_path=/tmp/deployment/application/node_modules/bcrypt/lib/binding"gyp ERR! cwd /tmp/deployment/application/node_modules/bcryptgyp ERR! node -v v10.13.0gyp ERR! node-gyp -v v3.8.0gyp ERR! not ok node-pre-gyp ERR! build error node-pre-gyp ERR! stack Error: Failed to execute '/opt/elasticbeanstalk/node-install/node-v10.13.0-linux-x64/bin/node /opt/elasticbeanstalk/node-install/node-v10.13.0-linux-x64/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/tmp/deployment/application/node_modules/bcrypt/lib/binding/bcrypt_lib.node --module_name=bcrypt_lib --module_path=/tmp/deployment/application/node_modules/bcrypt/lib/binding' (1)node-pre-gyp ERR! stack at ChildProcess.<anonymous> (/tmp/deployment/application/node_modules/node-pre-gyp/lib/util/compile.js:83:29)node-pre-gyp ERR! stack at ChildProcess.emit (events.js:182:13)node-pre-gyp ERR! stack at maybeClose (internal/child_process.js:962:16)node-pre-gyp ERR! stack at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)node-pre-gyp ERR! System Linux 4.14.77-69.57.amzn1.x86_64node-pre-gyp ERR! command "/opt/elasticbeanstalk/node-install/node-v10.13.0-linux-x64/bin/node" "/tmp/deployment/application/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"node-pre-gyp ERR! cwd /tmp/deployment/application/node_modules/bcryptnode-pre-gyp ERR! node -v v10.13.0node-pre-gyp ERR! node-pre-gyp -v v0.6.36node-pre-gyp ERR! not ok Failed to execute '/opt/elasticbeanstalk/node-install/node-v10.13.0-linux-x64/bin/node /opt/elasticbeanstalk/node-install/node-v10.13.0-linux-x64/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/tmp/deployment/application/node_modules/bcrypt/lib/binding/bcrypt_lib.node --module_name=bcrypt_lib --module_path=/tmp/deployment/application/node_modules/bcrypt/lib/binding' (1)CXX(target) Release/obj.target/bcrypt_lib/src/bcrypt_node.o../src/bcrypt_node.cc:8:22: fatal error: node_blf.h: No such file or directorycompilation terminated.make: *** [Release/obj.target/bcrypt_lib/src/bcrypt_node.o] Error 1make: Leaving directory `/tmp/deployment/application/node_modules/bcrypt/build'gyp ERR! build error gyp ERR! stack Error: `make` failed with exit code: 2gyp ERR! stack at ChildProcess.onExit (/opt/elasticbeanstalk/node-install/node-v10.13.0-linux-x64/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)gyp ERR! stack at ChildProcess.emit (events.js:182:13)gyp ERR! stack at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)gyp ERR! System Linux 4.14.77-69.57.amzn1.x86_64gyp ERR! command "/opt/elasticbeanstalk/node-install/node-v10.13.0-linux-x64/bin/node" "/opt/elasticbeanstalk/node-install/node-v10.13.0-linux-x64/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/tmp/deployment/application/node_modules/bcrypt/lib/binding/bcrypt_lib.node" "--module_name=bcrypt_lib" "--module_path=/tmp/deployment/application/node_modules/bcrypt/lib/binding"gyp ERR! cwd /tmp/deployment/application/node_modules/bcryptgyp ERR! node -v v10.13.0gyp ERR! node-gyp -v v3.8.0gyp ERR! not ok

**Comment by lajpatshah** [comment]
Hey @agathver thanks for your help. Somehow AWS did weird thing of trying to install bcrypt 1.0.3 on node 10.13.0 instead of installing on node 6.14.4. 

I checked full log in depth and found that there was configuration error on AWS side while picking up the values instead of me providing it with right values. In short in managed updates which happens automatically it was picking up value 10.13.0 instead of 6.14.4. Only bcrypt did not installed on this config and my new instance never passed the health check. 

I am now migrating to node 10.13.0 along with bcrypt 3.0.2. I tested and now it is working. 

Thanks for your help
